src/plugins/intel_cpu/src/infer_request.h

```diff
@@ -4,106 +4,94 @@
-    : IInferRequestInternal(inputs, outputs), execNetwork(execNetwork_) {}
+    void create_infer_request();
+    InferenceEngine::Precision norm_to_input_supported_prec(const std::pair<const std::string, ov::Tensor>& input) const;
+    void pushInput(const std::string& inputName, ov::Tensor& inputBlob, InferenceEngine::Precision dataType);
 
-    void CreateInferRequest();
-    InferenceEngine::Precision normToInputSupportedPrec(const std::pair<const std::string, InferenceEngine::Blob::Ptr>& input) const;
-    void pushInput(const std::string& inputName, InferenceEngine::Blob::Ptr& inputBlob, InferenceEngine::Precision dataType);
-
-    virtual void initBlobs() = 0;
-    virtual void PushInputData() = 0;
+    void init_tensor(const std::string& name);
+    void push_input_data();
 
     Graph* graph = nullptr;
-    std::unordered_map<std::string, void*> externalPtr;
+    mutable std::unordered_map<std::string, void*> external_ptr;
 
 private:
-    void PushStates();
-    void PullStates();
-    void redefineMemoryForInputNodes();
-
-    std::shared_ptr<ExecNetwork>        execNetwork;
-    openvino::itt::handle_t             profilingTask;
-    std::vector<std::shared_ptr<InferenceEngine::IVariableStateInternal>> memoryStates;
-    AsyncInferRequest*                  _asyncRequest = nullptr;
+    void push_states();
+    void pull_states();
+    void redefine_memory_for_input_nodes();
+
+    void update_external_inputs();
+    InferenceEngine::TensorDesc create_tensor_desc(const ov::Tensor& tensor);
+
+    // Transformation shouldn't change model's input/output's precision, but actually it does.
+    // Some additional methods will handle it.
+    const ov::Output<const ov::Node>& get_internal_port(const ov::Output<const ov::Node>& port) const;
+    ov::Tensor get_port_tensor(const ov::Output<const ov::Node>& port) const;
+    // Check this port is original model port or compiled model's port, return true for compiled model's port
+    // This is because if precision has been changed in compiled model, it need distinguish them
+    bool check_compiled_model_port(const ov::Output<const ov::Node>& port) const;
+    mutable std::unordered_map<std::string, ov::Output<const ov::Node>> m_orig_ports_map;
+    // Store external tensor due to precision changes
+    mutable std::unordered_map<std::string, ov::Tensor> m_aux_tensors;
+    mutable std::unordered_map<std::string, bool> m_port_precision_changed;
+    bool m_is_legacy_api = false;
+
+    std::shared_ptr<const CompiledModel> m_compiled_model;
+    openvino::itt::handle_t m_profiling_task;
+    std::vector<std::shared_ptr<ov::IVariableState>> m_memory_states;
+    AsyncInferRequest* m_asyncRequest = nullptr;
+
+    mutable std::unordered_map<std::string, ov::Output<const ov::Node>> m_input_ports_map;
+    mutable std::unordered_map<std::string, ov::Output<const ov::Node>> m_output_ports_map;
+    std::unordered_map<std::string, ov::Tensor> m_outputs;
```

Comment:
Since we have m_tensors, why do we still keep m_outputs? Can it be optimized?

Reply:
`m_tensors` stores all tensors of infer_request's input and output data, it is map data structure in interface class 
 ISyncInferRequest, while `m_outputs` is used for graph's output, which may point to the same tensor._impl with some tensor in `m_tensors`. They have different purpose, maybe can be merged together if refactor some method in plugin api 2.0

Reply:
Actually the whole WA with `aux_tensor` doesn't look obviously necessary. We can use m_tensor container with tensors corresponding to the inputs and outputs the user work with (from the original model). If there is mismatch between tensors on the CPU graph inputs and outputs, then the precision conversion happens (as it worked before).

Reply:
`m_tensors` is defined in ISyncInferRequest class:

`mutable std::unordered_map<std::shared_ptr<ov::descriptor::Tensor>, ov::Tensor> m_tensors;`

It need match it's port - precision/shape, which is bind to compiled_model, if precision changes between original model and compiled model, we should create new tensor to store the original model precision data rather than reuse m_tensors. 

ov::ISyncInferRequest::m_tensors  --> compiled_model input/output tensors
aux_tensors                                   --> original model input/output tensors

If there is no precision changes, there will no aux_tensor.
We use aux_tensor to avoid frequently allocate/deallocate for precision conversion.

Reply:
What prevents us from initializing m_tensors with Tenosr objects of correct type? To my understanding, the user should work with only one set of input/output tensors, which correspond with the original model. So any inconsistency between the original model and the transformed model and the cpu graph shall be handled internally, completely hidden from the user. Thus, we can simply keep the correct set of tensors in `m_tensors`, here "correct" means the configurations of tensors consistent with the initial model. All the differences between the `m_tensors` and memory on the cpu graph I/O edges are handled on the infer_request and cpu graph levels. Up to this point I do not seen any need to use `aux_tesnsors`, they only increase the complexity of the solution.

Reply:
New plugin API definition binds m_tensors and input/output port together, the port is compiled model's port rather than original model's input/output port, and new plugin API requires that original model and compiled model should have the same precision input/output port, current cpu plugin cannot fully meet this requirement, so we need some memory (such as aux tensor) to store original model port's precision data, and graph infer is based on compiled model, so its input/output should be m_tensor.

Assume input/output port's precision both changed:

For input port:
      
        user -> set_tensor -> aux_tensor -> convert precision -> input tensor of m_tensors -> infer

For output port:
  
         infer -> internal memory or m_out shared -> convert precision ->aux_tensor --> user

You are right, user should work with one set of input/output tensors, which is original model port's tensor. Current implement does provide the same set of tensor if user call get_tensor() API, there is no inconsistency.
But there are some CI test which may use compiled model's port as input argument to set_tensor(), so we also handle such cases of set_tensor:

`void InferRequest::set_tensor(const ov::Output<ov::Node>& port, const Tensor& tensor) `

1)  original model's port + tensor of original model port's precision
2)  original model's port + tensor of compiled model port's precision
3)  compiled model's port + tensor of original model port's precision
4)  compiled model's port + tensor of compiled model port's precision

![image](https://github.com/openvinotoolkit/openvino/assets/31196718/0ca4542e-4922-4a2c-84a3-caa420c9edd8)